Cubical/HITs/Truncation/FromNegOne/Properties.agda

```diff
@@ -90,93 +99,347 @@ isOfHLevel→isSphereFilled {A = A} (ℕ→ℕ₋₁ n) h = helper λ x y → is
+isModalIsProp (HLevelTruncModality n) = isPropIsOfHLevel n
+◯             (HLevelTruncModality n) = hLevelTrunc n
+◯-isModal     (HLevelTruncModality n) = isOfHLevelTrunc n
+η (HLevelTruncModality zero) _ = tt*
+η (HLevelTruncModality (suc n)) = ∣_∣
+◯-elim (HLevelTruncModality zero) cB _ tt* = cB tt* .fst
+◯-elim (HLevelTruncModality (suc n)) = elim
+◯-elim-β (HLevelTruncModality zero) cB f a = cB tt* .snd (f a)
+◯-elim-β (HLevelTruncModality (suc n)) = λ _ _ _ → refl
+◯-=-isModal (HLevelTruncModality zero) x y = (isOfHLevelUnit* 1 x y) , (isOfHLevelUnit* 2 x y _)
+◯-=-isModal (HLevelTruncModality (suc n)) = isOfHLevelPath (suc n) (isOfHLevelTrunc (suc n))
 
-setTrunc≃Trunc0 : ∥ A ∥₀ ≃ ∥ A ∥ 0
-setTrunc≃Trunc0 =
-  isoToEquiv
-    (iso
-      (SetTrunc.elim (λ _ → isOfHLevel∥∥ 0) ∣_∣)
-      (ind (λ _ → squash₀) ∣_∣₀)
-      (ind (λ _ → isOfHLevelPath 2 (isOfHLevel∥∥ 0) _ _) (λ _ → refl))
-      (SetTrunc.elim (λ _ → isOfHLevelPath 2 squash₀ _ _) (λ _ → refl)))
+-- universal property
 
-groupoidTrunc≃Trunc1 : ∥ A ∥₁ ≃ ∥ A ∥ 1
-groupoidTrunc≃Trunc1 =
-  isoToEquiv
-    (iso
-      (GpdTrunc.elim (λ _ → isOfHLevel∥∥ 1) ∣_∣)
-      (ind (λ _ → squash₁) ∣_∣₁)
-      (ind (λ _ → isOfHLevelPath 3 (isOfHLevel∥∥ 1) _ _) (λ _ → refl))
-      (GpdTrunc.elim (λ _ → isOfHLevelPath 3 squash₁ _ _) (λ _ → refl)))
+univTrunc : ∀ {ℓ} (n : HLevel) {B : TypeOfHLevel ℓ n} → Iso (hLevelTrunc n A → B .fst) (A → B .fst)
+univTrunc zero {B , lev} = isContr→Iso (isOfHLevelΠ 0 (λ _ → lev)) (isOfHLevelΠ 0 λ _ → lev)
+Iso.fun (univTrunc (suc n) {B , lev}) g a = g ∣ a ∣
+Iso.inv (univTrunc (suc n) {B , lev}) = elim λ _ → lev
+Iso.rightInv (univTrunc (suc n) {B , lev}) b = refl
+Iso.leftInv (univTrunc (suc n) {B , lev}) b = funExt (elim (λ x → isOfHLevelPath _ lev _ _)
+                                                            λ a → refl)
+
+-- functorial action
+
+map : {n : HLevel} {B : Type ℓ'} (g : A → B)
+  → hLevelTrunc n A → hLevelTrunc n B
+map {n = zero} g = λ _ → tt*
+map {n = suc n} g = rec (isOfHLevelTrunc _) (λ a → ∣ g a ∣)
+
+mapCompIso : {n : HLevel} {B : Type ℓ'} → (Iso A B) → Iso (hLevelTrunc n A) (hLevelTrunc n B)
+mapCompIso {n = zero} {B} _ = isContr→Iso (isOfHLevelUnit* 0) (isOfHLevelUnit* 0)
+Iso.fun (mapCompIso {n = (suc n)} g) = map (Iso.fun g)
+Iso.inv (mapCompIso {n = (suc n)} g) = map (Iso.inv g)
+Iso.rightInv (mapCompIso {n = (suc n)} g) = elim (λ x → isOfHLevelPath _ (isOfHLevelTrunc _) _ _) λ b → cong ∣_∣ (Iso.rightInv g b)
+Iso.leftInv (mapCompIso {n = (suc n)} g) = elim (λ x → isOfHLevelPath _ (isOfHLevelTrunc _) _ _) λ a → cong ∣_∣ (Iso.leftInv g a)
+
+mapId : {n : HLevel} → ∀ t → map {n = n} (idfun A) t ≡ t
+mapId {n = 0} tt* = refl
+mapId {n = (suc n)} =
+  elim (λ _ → isOfHLevelPath (suc n) (isOfHLevelTrunc (suc n)) _ _) (λ _ → refl)
+
+-- equivalences to prop/set/groupoid truncations
+
+propTruncTrunc1Iso : Iso ∥ A ∥₁ (∥ A ∥ 1)
+Iso.fun propTruncTrunc1Iso = PropTrunc.elim (λ _ → isOfHLevelTrunc 1) ∣_∣
```

Comment:
Do you really need elim in the fun/inv cases and not just rec? Same for the other similar lemmas